src/plat/pc99/machine/acpi.c

```diff
@@ -244,20 +215,34 @@ acpi_table_init(void* entry, enum acpi_type table_type)
     return acpi_table;
 }
 
+BOOT_CODE acpi_rsdp_t*
+acpi_search_rsdp()
+{
+    char* addr;
+
+    for (addr = (char*)BIOS_PADDR_START; addr < (char*)BIOS_PADDR_END; addr += 16) {
+        if (strncmp(addr, acpi_str_rsd, 8) == 0) {
+            if (acpi_calc_checksum(addr, 20) == 0) {
+                printf("ACPI: RSDP paddr=%p\n", addr);
+                return acpi_table_init(addr, ACPI_RSDP);
+            }
+        }
+    }
+    return NULL;
+}
+
 BOOT_CODE acpi_rsdt_t*
-acpi_init(void)
+acpi_init(acpi_rsdp_t const * acpi_rsdp)
 {
-    acpi_rsdp_t* acpi_rsdp = acpi_get_rsdp();
     acpi_rsdt_t* acpi_rsdt;
     acpi_rsdt_t* acpi_rsdt_mapped;
 
-    if (acpi_rsdp == NULL) {
+    printf("ACPI: RSDP vaddr=%p\n", acpi_rsdp);
+
+    if (!acpi_rsdp || acpi_calc_checksum((char const *)acpi_rsdp, 20) != 0) {
         printf("BIOS: No ACPI support detected\n");
         return NULL;
     }
-    printf("ACPI: RSDP paddr=%p\n", acpi_rsdp);
-    acpi_rsdp = acpi_table_init(acpi_rsdp, ACPI_RSDP);
```

Comment:
I'm trying to understand this change. It seems like `acpi_search_rsdp` is returning the mapped rsdp address, but that only gets called from a multiboot1 boot. When using the value from multiboot2 as far as I can tell `acpi_table_init` does not get called, and we are just using the raw pointer. It isn't clear to me why that is guaranteed to be mapped in?

Reply:
> I'm trying to understand this change. It seems like acpi_search_rsdp is returning the mapped rsdp address, but that only gets called from a multiboot1 boot.

Yes.

> When using the value from multiboot2 as far as I can tell acpi_table_init does not get called, and we are just using the raw pointer. It isn't clear to me why that is guaranteed to be mapped in?

Multiboot 2 provides a  copy of the ACPI RSDP data, directly embedded within the already mapped Multiboot 2 structure. So, the pointer points into the dense packed MBI2 structure. (This seems in general the main difference to MBI1, where all parts of MBI information may lie in arbitrary memory locations connected by pointers.)

Reply:
Ahh I didn't pick this up from my quick skim of the multiboot2 spec. Thanks.